### ISSUE TITLE
[BUG]: the type `ExtractRequestBody` returns `never`

### ISSUE THREAD

**Issue by wolfy1339** [octokit/types.ts#528]
### What happened?

While updating `@octoit/plugin-rest-endpoint-methods` to use the latest release of this package, in https://github.com/octokit/plugin-rest-endpoint-methods.js/pull/632, the tests are erroring out.

I managed to narrow it down to the `ExtractRequestBody` type, as can be seen in the code snippet below

https://github.com/octokit/types.ts/blob/e7956808d0e1d019c3f6bc7a96d29468dc16ff42/src/generated/Endpoints.ts#L21-L37

### Versions

`@octokit/types` v9.1.2
`typescript` v5.0.4

### Relevant log output

The following error is for the `PATCH /repos/{owner}/{repo}/labels/{name}` endpoint. Every error is similar to it

```
Argument of type '{ owner: string; repo: string; name: string; color: string; data: string; }' is not assignable to parameter of type 'RequestParameters & Omit<{ owner: string; repo: string; name: string; } & { data: never; }, "baseUrl" | "headers" | "mediaType">'.
  Type '{ owner: string; repo: string; name: string; color: string; data: string; }' is not assignable to type 'Omit<{ owner: string; repo: string; name: string; } & { data: never; }, "baseUrl" | "headers" | "mediaType">'.
    Types of property 'data' are incompatible.
      Type 'string' is not assignable to type 'never'.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by wolfy1339** [comment]
There seems to be an issue with TypeScript not liking the indexes we pass

Here's a [Playground Link](https://www.typescriptlang.org/play?pretty=true#code/JYWwDg9gTgLgBAbzmAhjAFgZzgXzgMyghDgCIABCAYxggGtgYB6CMAUwDsUxgBaGAJ7tMpANwAocYPbI06OAF5ZGTAG1STKG0iYmCCAHcObKDj1bIZgEZQUHKuja6ENuw7NgiMNjWAQOmmwAjgCuwFoAJgD6mDBoIZhRDj50ulT+3gAeMCIAuqoA5BFsADZs3gW5EtJsyugAQhARAopkWqFOMI3NpHBs2ZwR2HRsAhD4dXAA-GTpHN7zvf0LQ3AjYxOoGOrtIZ3dAqS502TcYCXAVGh+AQBWmP5LAxyr6+N1O8F7sQdH6nMLGBHE5bdCfDo-JqHfKkAGcIEws4XK4wG5Me6PY4ALkQ4jg+LgETQKBxCDxBIpqgA0nBgBw1qN3qDwd8ulC-rCMvCjrkcczSLt9uyYXDFvkqVVyficKo3ps5CyhT0RVyxZK8DjSEjLtd-OiHhwnithoz5dsBV8ldCQQqLRC2cr1NqUWiMYbsbiKUS4qSpRS4NTafS5R87azfrleXAACqKyGOiUSCkykP8wXx6HquCknCiIA) demonstrating this issue

**Comment by wolfy1339** [comment]
The problem is that `openapi-typescript` generates the `requestBody` as potentially `undefined`, which causes issues since we assume that it's never `undefined`.

A simple workaround is to add a `Required<T>` to the type before passing it to `ExtractRequestBody<T>`

**Comment by wolfy1339** [comment]
This seems like intended behaviour, as some endpoints accept empty request bodies

**Comment by jakebailey** [comment]
Maybe my issue turned out to be different, but I though it was the same; I updated to 9.1.3 but still get an error reltaed to this `data` field being `never`:

```
scripts/open-user-pr.mjs:46:41 - error TS2345: Argument of type '{ owner: string; repo: string; pull_number: number; reviewers: string[]; }' is not assignable to parameter of type 'RequestParameters & Omit<{ owner: string; repo: string; pull_number: number; } & { data: never; }, "baseUrl" | "headers" | "mediaType">'.
  Property 'data' is missing in type '{ owner: string; repo: string; pull_number: number; reviewers: string[]; }' but required in type 'Omit<{ owner: string; repo: string; pull_number: number; } & { data: never; }, "baseUrl" | "headers" | "mediaType">'.

 46         await gh.pulls.requestReviewers({
                                            ~
 47             owner: prOwner,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
 50             reviewers,
    ~~~~~~~~~~~~~~~~~~~~~~
 51         });
    ~~~~~~~~~

  node_modules/@octokit/types/dist-types/generated/Endpoints.d.ts:14:5
    14     data: {
           ~~~~
    'data' is declared here.

scripts/request-pr-review.mjs:42:54 - error TS2345: Argument of type '{ owner: any; repo: any; pull_number: number; reviewers: any[]; }' is not assignable to parameter of type 'RequestParameters & Omit<{ owner: string; repo: string; pull_number: number; } & { data: never; }, "baseUrl" | "headers" | "mediaType">'.
  Property 'data' is missing in type '{ owner: any; repo: any; pull_number: number; reviewers: any[]; }' but required in type 'Omit<{ owner: string; repo: string; pull_number: number; } & { data: never; }, "baseUrl" | "headers" | "mediaType">'.

 42     const response = await gh.pulls.requestReviewers({
                                                         ~
 43         owner: options.owner,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
 46         reviewers,
    ~~~~~~~~~~~~~~~~~~
 47     });
    ~~~~~

  node_modules/@octokit/types/dist-types/generated/Endpoints.d.ts:14:5
    14     data: {
           ~~~~
    'data' is declared here.


Found 2 errors in 2 files.

Errors  Files
     1  scripts/open-user-pr.mjs:46
     1  scripts/request-pr-review.mjs:42
```

You can see the error here: https://github.com/microsoft/TypeScript/actions/runs/4823551961/jobs/8592186478?pr=53966

We pinned this dep back in TypeScript and are hoping to be able to unpin it again.

Should I file a new issue? Or is it the same and this wasn't actually fixed?

**Comment by nickfloyd** [comment]
Hey @jakebailey, apologies for the trouble here.  I'm looking at this now to figure out what's happening.  It appears to be the same or similar issue.  We'll let you know - and if it ends up being different, I'll open a new tracking issue for you so we can get moving on a solution more quickly.

**Comment by nickfloyd** [comment]
I found an instance of what @wolfy1339 found in types.ts in the types reference in [plugin-rest-endpoint-method.js](https://github.com/octokit/plugin-rest-endpoint-methods.js/blob/openapi-update/node_modules/@octokit/types/dist-types/generated/Endpoints.d.ts#L18) I'm going to have a closer look at @octokit/types.ts to see if I can find any more situations where this type of thing might occur.  I feel like @wolfy1339's solution is solid, It may just need to be propagated to other areas as well.

**Comment by nickfloyd** [comment]
So it appears that generation never happened (for whatever reason) so when [this](https://github.com/octokit/types.ts/blob/main/scripts/update-endpoints/templates/endpoints.ts.template#L38) got merged in, [this](https://github.com/octokit/types.ts/blob/main/src/generated/Endpoints.ts#L38) was never updated.  I'm trying to determine why now.

**Comment by nickfloyd** [comment]
Ok.. it looks like things are held up on the octokit/openapi side (one of the [workflows](https://github.com/octokit/openapi/pull/347) is not running. Let me see if I can get that unblocked and hopefully, that will get things sorted here.

**Comment by wolfy1339** [comment]
The openapi side shouldn't affect the types package.
Generating that file and releasing it should help.

If it still happens, I'll have a look this weekend

**Comment by nickfloyd** [comment]
> The openapi side shouldn't affect the types package.
[This](https://github.com/octokit/types.ts/blob/main/scripts/update-endpoints/generated/README.md#L3) was conflating my thinking around that but I am glad that's not the case.  Thanks for the clarity!

**Comment by jakebailey** [comment]
Thanks, that did seem to fix the errors I listed above, but, now there's new ones. I'll make a new issue.

**Comment by wolfy1339** [comment]
Please wait for #533 to be released here and downstream, before making another issue please.

I'm going to merge them ASAP

**Comment by jakebailey** [comment]
Ah, sorry, I was too quick and filed https://github.com/octokit/types.ts/issues/534.

I'll wait a bit more 😄
